### PR TITLE
Cycles lights

### PR DESCRIPTION
**Pull request by rcorre**
Implement naive handling of cycles lights.
I don't try to handle any fancy node tree setups, but this will at least export the basic cycles light settings available in the side panel (color/strength/shadow).

### CODE REVIEW

File: tests/reference_exports/light/animation_various_lights.escn [hunk 1]
```diff
@@ -56,7 +56,7 @@ tracks/0/keys = {
 	"times":PoolRealArray(0.0, 0.833333, 1.25, 1.91667),
 	"transitions":PoolRealArray(1, 1, 1, 1),
 	"update":0,
-	"values":[true, false, true, true]
+	"values":[true, true, true, true]
```
Comment (rcorre) [inline tests/reference_exports/light/animation_various_lights.escn]: This is unrelated to my light changes, and I get this diff even on master. I'm guessing that either this reference scene wasn't updated in the past, or there's some difference with my blender version.

Comment (Jason0214) [reply tests/reference_exports/light/animation_various_lights.escn]: I think it is caused by you moved 
```
        light_node['shadow_enabled'] = light.shadow_method != "NOSHADOW"
        light_node['light_negative'] = light.use_negative
```
to
```
        AttributeConvertInfo(
            'shadow_method', 'shadow_enabled', lambda x: x != "NOSHADOW"
        ),
        AttributeConvertInfo('use_negative', 'light_negative', lambda x: x),
```
The `AttributeConvertInfo` is also used for animation exporting, the lambda also convert animation value from blender to godot.   The  `shadow_enabled` and `light_negative` in animation can not be handled the same way as the node attribute, so they are previously not written as `AttributeConvertInfo`.  I think you'd better keep what it was to avoid breaking this example.

File: io_scene_godot/converters/simple_nodes.py [hunk 2]
```diff
@@ -115,23 +140,25 @@ def export_lamp_node(escn_file, export_settings, node, parent_gd_node):
...
+        if not (light.use_sphere or light.use_nodes):
             logging.warning(
                 "Ranged light without sphere enabled: %s", node.name
             )
 
     elif light.type == "SPOT":
-        light_node = LightNode(node.name, 'SpotLight', parent_gd_node)
-        if not light.use_sphere:
+        light_node = klass(node.name, 'SpotLight', parent_gd_node)
+        if not (light.use_sphere or light.use_nodes):
```
Comment (Jason0214) [inline io_scene_godot/converters/simple_nodes.py]: I prefer here add a line of comment says cycle light has no `use_sphere`, and do you think it is logically more clear to write it as `if not light.use_nodes and not light.use_sphere`?

File: io_scene_godot/converters/simple_nodes.py [hunk 3]
```diff
@@ -115,23 +140,25 @@ def export_lamp_node(escn_file, export_settings, node, parent_gd_node):
...
     light = node.data
 
+    klass = CyclesLightNode if light.use_nodes else LightNode
+
     if light.type == "POINT":
-        light_node = LightNode(node.name, 'OmniLight', parent_gd_node)
+        light_node = klass(node.name, 'OmniLight', parent_gd_node)
 
-        if not light.use_sphere:
+        if not (light.use_sphere or light.use_nodes):
```
Comment (Jason0214) [inline io_scene_godot/converters/simple_nodes.py]: same as line below

File: io_scene_godot/converters/simple_nodes.py [hunk 4]
```diff
@@ -145,8 +172,13 @@ def export_lamp_node(escn_file, export_settings, node, parent_gd_node):
 
         # Properties common to all lights
         light_node['transform'] = fix_directional_transform(node.matrix_local)
```
Comment (Jason0214) [inline io_scene_godot/converters/simple_nodes.py]: could you do me a favor adding a comment here just mention these two attributes are specially handled, because they are not the same format in node attribute and animation value, really appreciate it!

File: io_scene_godot/converters/simple_nodes.py [hunk 5]
```diff
@@ -106,6 +116,21 @@ def attribute_conversion(self):
         return self._light_attr_conv
 
 
+class CyclesLightNode(LightNode):
+    """Base class for godot light node based on a cycles engine light."""
+    _light_attr_conv = [
```
Comment (Jason0214) [inline io_scene_godot/converters/simple_nodes.py]: I just checked the animation of cycles light, it is no trivial to handle, so `AttributeConvertInfo` has no help. Better to handle this two attributes as what we did to `shadow_enabled`, which is more readable.  Therefore, this new class may not be necessary.

File: io_scene_godot/converters/simple_nodes.py [hunk 6]
```diff
@@ -72,6 +72,16 @@ def export_camera_node(escn_file, export_settings, node, parent_gd_node):
...
 
 
+def emission_input(node_tree, name):
+    """Get the input from the emission shader node with the given name."""
+    for node in node_tree.nodes:
+        if node.bl_idname == 'ShaderNodeEmission':
+            for input in n.inputs:
+                if input.name == name:
+                    return input.default_value
+    return None
```
Comment (Jason0214) [inline io_scene_godot/converters/simple_nodes.py]: Better to return some default value like `100` and throw a warning, otherwise a node tree with no `ShaderNodeEmission` would crash the program

File: io_scene_godot/converters/simple_nodes.py [hunk 7]
```diff
@@ -72,6 +72,16 @@ def export_camera_node(escn_file, export_settings, node, parent_gd_node):
     return cam_node
 
 
+def emission_input(node_tree, name):
+    """Get the input from the emission shader node with the given name."""
+    for node in node_tree.nodes:
+        if node.bl_idname == 'ShaderNodeEmission':
+            for input in n.inputs:
```
Comment (Jason0214) [inline io_scene_godot/converters/simple_nodes.py]: Here is what CI reported, seems it something left out in your last commit fixing pylint error.   Sometimes the pylint get quite annoying, however it do helps a lot to format the code :)

**Comment by Jason0214** [comment]
overall great work!  Really appreciate your effort!

**Comment by Jason0214** [comment]
looks really good now! fix the test and squash commits and I will merge it.

**Comment by rcorre** [comment]
Done. I left the `gamma_correct` RGBA workaround as a separate commit as I thought it was a distinct piece of work, but I can squash that too if you really want a single commit.

**Comment by Jason0214** [comment]
I also think it better to keep the `gamma_correct` fix as separate commit. Merge it!
